### PR TITLE
Adds basic content component support

### PR DESCRIPTION
**Pull request by aduric**
Hey guys, my first attempt at anything Polymer and Webcomponent and HATEOAS (all together). I managed to hack up a basic content component, but after rebasing recently, the sub-components do not load. Am I missing something else?

### CODE REVIEW

File: data/items.js [hunk 1]
```diff
@@ -261,6 +261,12 @@ window.D2L.EntityStore.update('https://f1bbe417-8c9c-48c0-90b2-75c0cfaceda5.user
 				'https://api.brightspace.com/rels/updates'
 			],
 			'href': 'https://f1bbe417-8c9c-48c0-90b2-75c0cfaceda5.updates.api.proddev.d2l/feed?filter=feed&count=20&page=1'
+		},
+		{
+			'rel': [
+				'https://api.brightspace.com/rels/content'
+			],
+			'href': 'https://f1bbe417-8c9c-48c0-90b2-75c0cfaceda5.sequences.api.proddev.d2l/6609'
```
Comment (awikkerink) [inline data/items.js]: The link for a sequence is inside the organization entity. Not the user entity: http://search.dev.d2l/source/xref/Lms/core/le/content/D2L.LE.Content/Extensibility/HM/Content/Plugins/ContentOrganizationalLinkPlugin.cs

File: src/content/content-topic-list.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+					'_changed(entity)'
+				];
+			}
+
+			_changed(entity) {
+				this.items = entity.entities;
+			}
+
+			getHref(item) {
+				return item.links[0].href;
```
Comment (awikkerink) [inline src/content/content-topic-list.html]: Use `item.getLinkByRel`

File: src/content/content-topic-list.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+			}
+
+			static get observers() {
+				return [
+					'_changed(entity)'
+				];
+			}
+
+			_changed(entity) {
+				this.items = entity.entities;
```
Comment (awikkerink) [inline src/content/content-topic-list.html]: Use `entity.getSubEntitiesByRel`

File: src/content/content-module-list.html [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+			static get observers() {
+				return [
+					'_changed(entity)'
+				];
+			}
+
+			_changed(entity) {
+				this.items = entity.entities;
+				console.log(this.items);
```
Comment (awikkerink) [inline src/content/content-module-list.html]: *nit* `console.log`

File: src/content/content-topic-list.html (hunk 3 above)
Comment (awikkerink) [reply src/content/content-topic-list.html]: ditto for the other places entity.entities is used

File: src/content/content-topic-card.html [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+			}
+
+			static get observers() {
+				return [
+					'_changed(entity)'
+				];
+			}
+
+			_changed(entity) {
+			}
```
Comment (awikkerink) [inline src/content/content-topic-card.html]: Remove empty observer

File: src/content/content-module-card.html [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+	<script>
+		/* @mixes SirenEntityMixin */
+		class ContentModuleCard extends SirenEntityMixin(Polymer.Element) {
+
+			static get is() { return 'd2l-content-module-card'; }
+
+			static get properties() {
+				return {
+					type: String
```
Comment (awikkerink) [inline src/content/content-module-card.html]: Is this used anywhere?

File: src/content/content-module-list.html [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+	<script>
+		class ContentModuleList extends EntityMixin(Polymer.Element) {
+
+			static get is() { return 'd2l-content-module-list'; }
+
+			static get properties() {
+				return {
+					title: String,
+					token: String,
+					href: String
```
Comment (awikkerink) [inline src/content/content-module-list.html]: No need to specify token/href.

You do need to put `/* @mixes EntityMixin */` above `class ContentModuleList`

File: src/content/content-topic-card.html [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+	<script>
+		/* @mixes SirenEntityMixin */
+		class ContentTopicCard extends SirenEntityMixin(Polymer.Element) {
+
+			static get is() { return 'd2l-content-topic-card'; }
+
+			static get properties() {
+				return {
+					type: String
```
Comment (awikkerink) [inline src/content/content-topic-card.html]: Is this used?

File: src/content/content-topic-list.html [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+			static get is() { return 'd2l-content-topic-list'; }
+
+			static get properties() {
+				return {
+					posts: {
+						type: Array,
+						value: []
+					},
+					token: String,
+					href: String
```
Comment (awikkerink) [inline src/content/content-topic-list.html]: No need to specify token/href.

File: src/content/content-topic-list.html [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+	<script>
+		/* @mixes SirenEntityMixin */
+		class ContentTopicList extends SirenEntityMixin(Polymer.Element) {
+
+			static get is() { return 'd2l-content-topic-list'; }
+
+			static get properties() {
+				return {
+					posts: {
```
Comment (awikkerink) [inline src/content/content-topic-list.html]: Is `posts` used anywhere?

File: src/content/content-module-card.html [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+<link rel="import" href="content-topic-list.html">
+
+<dom-module id="d2l-content-module-card">
+	<template>
+		<style>
+			:host {
+				display: block;
+			}
+		</style>
+		<paper-card style="width: calc(100% - 20px); margin: 10px; padding: 0;">
```
Comment (awikkerink) [inline src/content/content-module-card.html]: Use css instead of inline styles

Comment (aduric) [reply src/content/content-module-card.html]: Noted. For now I just copied stuff in the other components that seem to be using inline styles, so I will clean this up in another PR.

File: src/content/content-module-list.html [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+			</div>
+			<template is="dom-repeat" items="[[items]]">
+				<d2l-content-module-card href="[[getHref(item)]]" token="{{token}}"></d2l-content-module-card>
+			</template>
+		</paper-card>
+	</template>
+
+	<script>
+		/* @mixes SirenEntityMixin */
+		class ContentModuleList extends EntityMixin(Polymer.Element) {
```
Comment (awikkerink) [inline src/content/content-module-list.html]: This should be `SirenEntityMixin`.

Comment (aduric) [reply src/content/content-module-list.html]: Oh ya, sorry missed that

File: src/content/content-module-list.html [hunk 13]
```diff
@@ -0,0 +1,61 @@
+<link rel="import" href="../../bower_components/polymer/polymer-element.html">
+<link rel="import" href="../../bower_components/paper-card/paper-card.html">
+<link rel="import" href="../entity-mixin.html">
```
Comment (awikkerink) [inline src/content/content-module-list.html]: Should be `../siren-entity-mixin.html`

**Comment by aduric** [comment]
One thing I'm wondering about: If an Org doesn't have any child components of a type (let's say there not discussion forums for example) how do we _hide_ that component?

**Comment by awikkerink** [comment]
@aduric You can use `dom-if`:

```html
...
<template is="dom-if" if="{{discussions}}">
<d2l-discussions-forum-list href="{{discussions.href}}" token="{{token}}"></d2l-discussions-forum-list><br>
</template>
<template is="dom-if" if="{{content}}">
<d2l-content-module-list href="{{content.href}}" token="{{token}}"></d2l-content-module-list><br>
</template>
...
```

File: src/content/content-module-card.html [hunk 14]
```diff
@@ -0,0 +1,54 @@
+<link rel="import" href="../../bower_components/polymer/polymer-element.html">
+<link rel="import" href="../../bower_components/paper-card/paper-card.html">
+<link rel="import" href="../../bower_components/paper-button/paper-button.html">
+<link rel="import" href="../siren-entity-mixin.html">
+<link rel="import" href="content-topic-list.html">
```
Comment (awikkerink) [inline src/content/content-module-card.html]: This component doesn't exist

File: src/content/content-module-card.html [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+				<span>[[module.properties.title]]<span>
+			</div>
+			<div role="listbox">
+				<template is="dom-repeat" items="[[topics]]">
+					<paper-card style="width: calc(100% - 20px); margin: 10px; padding: 0px;">
+						<div style="font-weight:bold; padding: 10px 0px 0px 10px">[[item.properties.title]]</div>
+						<div style="background-color:lightgray;">
+							<template is="dom-repeat" items="[[item.entities]]">
+								<!-- sketchy array access pattern below -->
+								<a href="[[item.links.0.href]]">[[item.properties.title]]</a>
```
Comment (awikkerink) [inline src/content/content-module-card.html]: Maybe add a `_getLink` method that uses `getLinkByRel`.

File: src/content/content-module-card.html [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+			static get observers() {
+				return [
+					'_changed(entity)'
+				];
+			}
+
+			_changed(entity) {
+				this.module = entity;
+				this.topics = entity.entities;
```
Comment (awikkerink) [inline src/content/content-module-card.html]: I would prefer calling `getSubEntitiesByRel` to ensure all the entities are of a certain type.

File: src/content/content-module-list.html [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+					'_changed(entity)'
+				];
+			}
+
+			_changed(entity) {
+				this.items = entity.entities;
+			}
+
+			getHref(item) {
+				return item.links[0].href;
```
Comment (awikkerink) [inline src/content/content-module-list.html]: `getLinkByRel`

**Comment by aduric** [comment]
Closing this for now as I need to rethink sequence-activities a bit more. Thanks for the feedback thus far!

**Pull request closed without merging**